### ISSUE TITLE
Support the analysis of matched samples

### ISSUE THREAD

**Issue by antagomir** [microbiome/mia#196]
It is relatively common that an experimental study has paired samples, for instance baseline and intervention time points from the same individuals. Then one would prefer paired tests and visualizations. Potentially there could be also more time points, with such matching.

This could be supported by providing handy methods to split the `SE` object into such matched assays. The `altExp` structure could be suitable for this (this would allow having this simultaneously for multiple experiments in `MAE`, if necessary). If there is a separate time series structure available for SE/altExp/MAE, that would be even better but I am not aware of such solution. 

The method could be something like `splitByPair(se, pair_field="subject")` and this would split the original assay in `se` into multiple `altExp` assays. The columns should match between them (each column is one subject, and named accordingly). 

There are (at least) the following potential problems, however:

1) `colData` is per sample, and if we split to `altExps` where columns correspond to subjects (not samples), then we could loose the information that is sample-specific. How to retain the sample-specific information?

2) Some samples do not match (some subjects are missing some samples etc). Then the method could either discard the non-matched samples, or fill in `NA` for the missing data. Neither solution is ideal.

3) The altExp assays might have an order (for instance baseline to intervention; or more generally sequence across time). There would need to be a way to provide ordering for altExp tables but the structure does not naturally support this.

If generic solution for time series etc. seems too difficult, perhaps there could anyway be a simple method that allows splitting paired samples into altExp/MAE to support fast exploration.

**Comment by FelixErnst** [comment]
In my opinion you are thinking to complicated about this. 

The first step is to have one or in this case two factor(s) in the colData. Whether these factors are subjects, timepoint or who knows what is irrelenvant. If you implement a function which takes into consideration two factors, one for the matching and for the test, you are done, aren't you?

Any splitting is easy: just use `splitBy` as proposed in #191

**Comment by antagomir** [comment]
Yes, if we are ok with having the output from splitBy as a list, rather than use `altExp`, `MAE`, or other such mechanism and their associated methods.

**Comment by TuomasBorman** [comment]
The proposed solution with `splitOn()` works sufficiently. As it can be used for both columns and rows, I think it is better to always output a list for harmonization as the result cannot be added to `altExp` if splitting was done for columns.
